shark_turbine/kernel/wave/promotion.py

```diff
@@ -42,7 +42,7 @@ def promote_node(
     with node.graph.inserting_before(node.fx_node.next):
         constrained_shape = get_constrained_shape(node.type.symbolic_shape, constraints)
         allocate_node = Allocate(
-            node.memory_type.symbolic_shape,
+            node.type.symbolic_shape,
```

Comment:
Shouldn't this be memory type?

Reply:
For reads with mapping op sequence will look like this:
`global_mem -> gather -> write -> shared_mem alloc -> read -> mma`

`memory_type.symbolic_shape` will return shape of the gather input and `type.symbolic_shape` will be shape of gather output, what we really want here.

Reply:
What about without mapping op? Then the sequence would be
`global_mem -> write -> shared_mem alloc -> read -> mma`

What would `memory_type.symbolic_shape` and `type.symbolic_shape` return in this case?

Reply:
sounds good, thanks.